### PR TITLE
Feat/cart/cpc 1033/retrieve cart for client

### PR DESCRIPTION
**Pull request by Hares-2088**
JIRA: https://champlainsaintlambert.atlassian.net/browse/CPC-1033
## Context:
Retrieve the informations of a cart with all its products using a clientId
## Changes
 - created the cart entity
 - implemented the get cart by cart id
 - added a domain client layer for a product client
 - added some tests
 - modified the application.yaml to integrate a mongo Database
## Dev notes (Optional)
- Still need to add it to the API gateway

### CODE REVIEW

**Comment by NtsoaElie** [comment]
Everything looks fine to me.

**Review by NtsoaElie — APPROVED** [review]

**Review by NtsoaElie — APPROVED** [review]

**Review by NtsoaElie — APPROVED** [review]

File: cart-service/src/test/java/com/petclinic/cartsservice/dataaccesslayer/CartRepositoryUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+//        //act and assert
+//        StepVerifier
+//                .create(cartRepository.findCartByCartId(cart1.getCartId()))
+//                .consumeNextWith(foundCart -> {
+//                    assertNotNull(foundCart);
+//                    assertEquals(cart1.getCartId(), foundCart.getCartId());
+//                    assertEquals(cart1.getCustomerId(), foundCart.getCustomerId());
+//                })
+//                .verifyComplete();
+//    }
```
Comment (nic5694) [inline cart-service/src/test/java/com/petclinic/cartsservice/dataaccesslayer/CartRepositoryUnitTest.java:67]: You should not be pushing commented out code into main.

**Review by nic5694 — COMMENTED** [review]
Please check out the comment i left regarding the commented out test

File: cart-service/src/test/java/com/petclinic/cartsservice/dataaccesslayer/CartRepositoryUnitTest.java (hunk 1 above)
Comment (Hares-2088) [reply cart-service/src/test/java/com/petclinic/cartsservice/dataaccesslayer/CartRepositoryUnitTest.java:67]: Thank you. This will be fixed ASAP

**Review by Zako563 — APPROVED** [review]
Looks good to me. I don't see any code that can occur an error. It  mostly looks how the previous teams did.

**Pull request merged**